clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -793,7 +794,9 @@ public void seek(TopicPartition partition, long offset) {
                 offset,
                 Optional.empty(), // This will ensure we skip validation
                 metadata.currentLeader(partition));
-            subscriptions.seekUnvalidated(partition, newPosition);
+            Timer timer = time.timer(defaultApiTimeoutMs);
+            SeekUnvalidatedEvent seekUnvalidatedEventEvent = new SeekUnvalidatedEvent(calculateDeadlineMs(timer), partition, newPosition);
+            applicationEventHandler.addAndGet(seekUnvalidatedEventEvent);
```

Comment:
Per the other suggestions, we could defer the creation of the `SubscriptionState.FetchPosition ` until the `ApplicationEventProcessor` processor method to make this call a bit more succinct:

```suggestion
            SeekUnvalidatedEvent event = new SeekUnvalidatedEvent(
                calculateDeadlineMs(timer),
                partition,
                offset,
                Optional.empty(), // This will ensure we skip validation
            );
            applicationEventHandler.addAndGet(event);
```